ga4gh/datamodel/reads.py

```diff
@@ -91,6 +91,124 @@ def setFlag(flagAttr, flag):
         return flagAttr | flag
 
 
+class AlignmentDataMixin(datamodel.PysamDatamodelMixin):
+    """
+    Mixin class that provides methods for getting read alignments
+    from bam files
+    """
+    def _getReadAlignments(
+            self, reference, start=None, end=None, readGroup=None):
+        """
+        Returns an iterator over the specified reads
+        """
+        # TODO If reference is None, return against all references,
+        # including unmapped reads.
+        samFile = self.getFileHandle(self._samFilePath)
+        referenceName = reference.getLocalId().encode()
+        # TODO deal with errors from htslib
+        start, end = self.sanitizeAlignmentFileFetch(start, end)
+        readAlignments = samFile.fetch(referenceName, start, end)
+        for readAlignment in readAlignments:
+            tags = dict(readAlignment.tags)
+            if readGroup is None:
+                if 'RG' in tags:
+                    alignmentReadGroupId = tags['RG']
+                    alignmentReadGroup = self._readGroupLocalIdMap[
+                        alignmentReadGroupId]
+                    yield self.convertReadAlignment(
+                        readAlignment, alignmentReadGroup)
+                else:
+                    # TODO what to do for unmapped reads (i.e. no
+                    # alignment read group context)? skip for now
+                    pass
```

Comment:
In all cases, the readGroup is used in convertReadAlignment to set the read.readGroupId and the read.id (which is a compoundId that, of course, includes the readGroupId in its hierarchy).

To me, it sounds like we should not be setting these id(s) as we currently are in the case of unmapped reads (i.e. reads that have no readGroup).  What should we do then instead?  Keep in mind these ids need to be both unique and stable (i.e. we can't just use a random number).  What about:
```
datasetId:readGroupSetId:unmapped:readId
```